src/schematics/shared/get-file-contents.ts

```diff
@@ -0,0 +1,12 @@
+import {
+  Tree
+} from '@angular-devkit/schematics';
+
+export function getFileContents(tree: Tree, filePath: string): string {
+  const contentBuffer = tree.read(filePath);
+  if (!contentBuffer) {
+    throw new Error(`Cannot read "${filePath}" because it does not exist.`);
```

Comment:
Since this isn't reading against the file system, I doubt there would be a case where the file exists in the `Tree` but Angular can't read it (due to file permissions, etc.). I'm willing to come back to this if it becomes a problem, however.